benchtop/src/sp_trie.rs

```diff
@@ -106,39 +80,27 @@ impl Db for SpTrieDB {
         };
 
         let recorder: sp_trie::recorder::Recorder<Hasher> = Default::default();
-        let mut trie_recorder = recorder.as_trie_recorder(new_root);
-        let mut trie_db_mut = TrieDBMutBuilderV1::from_existing(&mut trie, &mut new_root)
-            .with_recorder(&mut trie_recorder)
-            .build();
-
-        for action in actions.into_iter() {
-            match action {
-                Action::Write { key, value } => {
-                    // sp_trie does not require hashed keys,
-                    // but if keys are not hashed, the comparison does not seem to be efficient.
-                    // Not applying hashing to keys would significantly speed up sp_trie.
-                    let key_path = sha2::Sha256::digest(key);
-
-                    trie_db_mut
-                        .insert(&key_path, &value.unwrap_or(vec![]))
-                        .expect("Impossible writing into sp-trie db");
-                }
-                Action::Read { key } => {
-                    let key_path = sha2::Sha256::digest(key);
-
-                    let _timer_guard_read = timer.as_mut().map(|t| t.record_span("read"));
-                    trie_db_mut
-                        .get(&key_path)
-                        .expect("Impossible fetching from sp-trie db");
-                }
-            }
-        }
+        let _timer_guard_commit = {
```

Comment:
why this scope?

Reply:
the scope is required for the borrow of `new_root` to end. moving the new_root borrow into the `Tx` struct confuses Rust's Non-Lexical Lifetime checker so we have to fall back to old-school lexical lifetimes :)